src/Microsoft.ML.Data/EntryPoints/PredictorModel.cs

```diff
@@ -36,6 +36,18 @@ public PredictorModel(IHostEnvironment env, RoleMappedData trainingData, IDataVi
             _predictor = predictor;
         }
 
+        //REVIEW: I'm not sure this is the right thing to do because we are setting predictor to null
+        //when this class is supposed to contain a predictor. TransformModel may or may not
+        //contain a predictor. Here we are just using this class as a wrapper for TransformModel 
+        //so that we can use a single class to accept TransformModel and PredictorModel has inputs.
+        public PredictorModel(IHostEnvironment env, ITransformModel transformModel)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(transformModel, nameof(transformModel));
+
+            _transformModel = transformModel;
+        }
```

Comment:
OK, so you had to change predictor model so that its prior invariant, that it always contains a predictor, no longer holds. This amounts to a fairly substantial reorganization -- all code against this structure would have been based on the prior assumption. :( My understanding was that this was already baked into the class, but that isn't right.

With that new information, we shouldn't do that. I'd rather just have the entry-point take an `ITransformModel` only. :P At a later time as a separate work item, if needed, we can introduce an entry-point that takes a predictor model and produces a transform model out of it.

Reply:
I'm glad....

---
In reply to: [191834251](https://github.com/dotnet/machinelearning/pull/248#discussion_r191834251) [](ancestors = 191834251)